### PR TITLE
[Merged by Bors] - Update golangci-lint and fix findings

### PR DESCRIPTION
**Pull request by fasmat**
## Motivation

This updates `golangci-lint` to the newest version (and updated config) and fixes all reported findings.

## Description

This is a pre-requisite for https://github.com/spacemeshos/go-spacemesh/pull/5580 as the old `golangci-lint` version isn't fully compatible with Go 1.22 yet.

## Test Plan

- existing tests pass

## TODO

- [x] Explain motivation or link existing issue(s)
- [x] Test changes and document test plan
- [x] Update documentation as needed
- [x] Update [changelog](../CHANGELOG.md) as needed

### CODE REVIEW

File: activation/activation_test.go [hunk 1]
```diff
@@ -862,8 +862,9 @@ func TestBuilder_PublishActivationTx_NoPrevATX_PublishFails_InitialPost_preserve
 		Challenge:     shared.ZeroChallenge,
 		LabelsPerUnit: tab.conf.LabelsPerUnit,
 	}
-	tab.mValidator.EXPECT().Post(gomock.Any(), sig.NodeID(), refPost.CommitmentATX, initialPost, meta, refPost.NumUnits).
-		Return(nil)
+	tab.mValidator.EXPECT().Post(
+		gomock.Any(), sig.NodeID(), refPost.CommitmentATX, initialPost, meta, refPost.NumUnits,
+	).Return(nil)
```
Comment (poszu) [inline activation/activation_test.go]: I find this more readable:
```suggestion
	tab.mValidator.EXPECT().
		Post(gomock.Any(), sig.NodeID(), refPost.CommitmentATX, initialPost, meta, refPost.NumUnits).
		Return(nil)
```
wdyt?

File: activation/interface.go [hunk 2]
```diff
@@ -80,8 +80,8 @@ type layerClock interface {
 }
 
 type nipostBuilder interface {
-	BuildNIPost(ctx context.Context, sig *signing.EdSigner, challenge *types.NIPostChallenge) (*nipost.NIPostState, error)
-	Proof(ctx context.Context, nodeID types.NodeID, challenge []byte) (*types.Post, *types.PostInfo, error)
+	BuildNIPost(context.Context, *signing.EdSigner, *types.NIPostChallenge) (*nipost.NIPostState, error)
+	Proof(context.Context, types.NodeID, []byte) (*types.Post, *types.PostInfo, error)
```
Comment (poszu) [inline activation/interface.go]: I'm not sure about this. Previously it was easy to understand what the 3rd parameter of `Proof()` is by looking at its name. Now, it's impossible.

Comment (fasmat) [reply activation/interface.go]: Yeah I wasn't sure how to best fix this. I guess line breaks after commas work too

**Review by poszu — APPROVED** [review]

**Comment by fasmat** [comment]
bors merge

**Comment by fasmat** [comment]
bors merge

**Pull request closed without merging**